src/rev_ai/apiclient.py

```diff
@@ -0,0 +1,130 @@
+
+class RevAiAPIClient:
+    """Client which implements rev.ai API"""
+
+    version = "v1"
+    """Default version of rev.ai"""
+
+    base_url = "https://api.rev.ai/revspeech/{v}/".format(v=version)
+    """Default address of the API"""
+
+    def __init__(self, api_key):
+        """Constructor
+
+        :param api_key: api key which authorizes all requests and links them to your account.
+            Generated on the settings page of your account dashboard on rev.ai
+        """
+        if not api_key:
+            raise ValueError("API Key cannot be empty.")
+        self.s = requests.Session()
+        self.s.headers.update({
+            'Authorization': 'Bearer {api_key}'.format(api_key=api_key)
+        })
+
+
+    def submit_job_url(self, media_url, options):
+        """Submit media given a URL for transcription.
+        The audio data is downloaded from the URL.
+
+        :param media_url: web location of the media file
+        :param options: JobSubmitOptions object for the job
+        :returns: raw response data
+        """
+        url_jobs = urljoin(self.base_url, "jobs")
+        payload = {
+            'media_url': media_url,
+            'metadata': options.metadata
+        }
+        if options.callback_url:
+            payload['callback_url'] = options.callback_url
+        response = self.s.post(url_jobs, json=payload)
+
+        return Job.from_json(response.json())
+
+    def submit_job_local_file(self, filename, options):
+        """Submit a local file for transcription.
+        Note that the content type is inferred if not provided.
+
+        :param filename: path to a local file on disk
+        :param content_type: MIME content type of file being sent
+        :param options: JobSubmitOptions object for the job
+        :returns: raw response data
+        """
+        url_jobs = urljoin(self.base_url, "jobs")
+        payload = {'metadata': options.metadata}
+        if options.callback_url:
+            payload['callback_url'] = options.callback_url
+
+        with open(filename, 'rb') as f:
+            files = {
+                'media': (filename, f),
+                'options': (None, json.dumps(payload))
+            }
+            response = self.s.post(url_jobs, files=files)
+
+        return Job.from_json(response.json())
+
+    def get_job_details(self, id_):
+        """View information about a specific job.
+        The server will respond with the status and creation date.
+
+        :param id_: id that the server gave you
+        :returns: raw response data
+        """
+        url_jobs_id = urljoin(self.base_url, "jobs/{id_}".format(id_=id_))
+
+        response = self.s.get(url_jobs_id)
+
+        return Job.from_json(response.json())
+
+    def get_transcript(self, id_, format_):
+        """Get the transcript of a specific job as json.
+
+        :param id_: id that the server gave you
+        :param format_: format of returned transcript, either "text" or "json"
```

Comment:
I'm not the biggest fan of magic string inputs. Documentation helps, but better is to have the function signature make things obvious. My personal choice would be a `fmt_text=True` flag (which wouldn't work if you have more than two formats). If more than two formats are planned, I would make two methods now.

Python enums suck.

Reply:
😢Makes sense, I was hoping to have an enum and it sucked. If we're going to have a boolean I would rather just have 2 methods